tests/zquantum/core/circuits/conversions/qiskit_conversions_test.py

```diff
@@ -108,7 +96,46 @@ def test_matrices_are_equal_up_to_phase_factor(self, theta, phi, lambda_):
         ).astype(np.complex128)
         qiskit_matrix = qiskit.extensions.U3Gate(theta, phi, lambda_).to_matrix()
 
-        assert _is_scaled_identity(zquantum_matrix @ np.linalg.inv(qiskit_matrix))
+        np.testing.assert_allclose(zquantum_matrix, qiskit_matrix, atol=1e-7)
+
+
+def _reorder_gate(G, perm):
+    perm = list(perm)
+
+    # number of qubits
+    n = len(perm)
+
+    # reorder both input and output dimensions
+    perm2 = perm + [n + i for i in perm]
+
+    return np.reshape(np.transpose(np.reshape(G, 2 * n * [2]), perm2), (2 ** n, 2 ** n))
```

Comment:
This function is rather hard to digest and takes a while to understand. What do you think about writing some docstring explaining what it does? Also, I recommend changing its name to `_reorder_gate_matrix` since it operates on matrices, not gates.